docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,90 @@
+[[runtime]]
+=== Runtime
+
+////
+[source,console]
+----
+PUT /my-index
+----
+// TESTSETUP
+////
+
+Typically, you must index fields to {es} before they can be retrieved,
+aggregated, or searched. With runtime fields, you can explicitly define a field
+in the mapping and access it at runtime without indexing your data. This
+flexibility allows you to more quickly ingest raw data into the Elastic Stack
+and immediately access it. By dynamically evaluating runtime fields at query
+time, you can choose which fields to index and optimize disk space.
+
+Runtime fields are not indexed, but are accessible from the search API like any
+other field that has <<doc-values,doc_values>> and is searchable. You can
+retrieve and query these fields, as well as aggregate and
+sort on them.
+
+NOTE: Runtime fields can be incur performance costs at search time, depending
+on the <<runtime-params-runtime-type,`runtime_type`>>.
+
+==== Specifying runtime fields
+You specify runtime fields in the mapping by
+<<modules-scripting-using,defining a script>>. At search time, the script runs
+and generates values for each scripted field.
+
+The script in the following request extracts the day of the week from the
+`timestamp` field, which is defined as a `date` data type.
```

Comment:
Mapping for other field types is almost always done upfront. You typically create a mapping as part of an index template or when you create an index. That means it happens before you start indexing data. To be useful, runtime fields are mapped after you start indexing data.

I think it'd be useful to show the user how the raw data is typically stored in ES before the runtime field is created. An in-depth example would be good. It may also be helpful to show an example of how a runtime field could replace an existing ingest pipeline, etc.